core/services/offchainreporting/block_translator.go

```diff
@@ -0,0 +1,79 @@
+package offchainreporting
+
+import (
+	"math/big"
+
+	"github.com/smartcontractkit/chainlink/core/chains"
+)
+
+// BlockTranslator converts emitted block numbers (from block.number) into a
+// block number range suitable for query in FilterLogs
+type BlockTranslator interface {
+	NumberToQueryRange(changedInBlock uint64) (fromBlock *big.Int, toBlock *big.Int)
+	Start()
+	Close()
+}
+
+// NewBlockTranslator returns the block translator for the given chain
+func NewBlockTranslator(chain *chains.Chain) BlockTranslator {
+	if chain.IsArbitrum() {
```

Comment:
Would it be better to do this with polymorphism? i.e.

```go
type arbitrumTranslator struct {}
func (a arbitrumTranslator) NewBlockTranslator(chain *chains.Chain) BlockTranslator { ... }
```

Reply:
Hmm, not sure I understand. Wouldn't we just move the `if` statement up one level to where we instantiate the `arbitrumTranslator`?